### PR TITLE
Add Transform for DetCon + MultiViewTransformV2 for torchvision.transforms.v2

### PR DESCRIPTION
**Pull request by liopeer**
## Changes
 - added `MultiViewTransformV2` as a new base class for all future transforms requiring `torchvision.transforms.v2`
 - added `DetConSimCLRViewTransform` as the single view transform for DetCon
 - added `DetConSTransform` that uses the previous `DetConSimCLRViewTransform` to generate 2 views
   - the applied transforms differ only in their application of Gaussian blurring
   - to compare the exact hyperparams from the model, check the paper ([DetCon](https://arxiv.org/pdf/2103.10957)) on page 9/13
   - add `MultiViewTransformV2` and `DetConSTransform` to docs
## How has it been tested?
 - testing correct tensor shapes
 - testing that masks are unaffected if user-provided
 - testing that masks are changed (and correct) if not user-provided
 ## Docs Screenshots (they do not show all the constructor arguments, but they are there!)
Screenshot 2024-11-18 at 16 34 12
Screenshot 2024-11-18 at 16 34 39

### CODE REVIEW

**Review by guarin — COMMENTED** [review]
Nice! Left minor comments.

File: lightly/transforms/detcon_transform.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from lightly.transforms.add_grid_transform import AddGridTransform
+from lightly.transforms.multi_view_transform_v2 import MultiViewTransformV2
+from lightly.transforms.torchvision_v2_compatibility import torchvision_transforms as T
+from lightly.transforms.utils import IMAGENET_NORMALIZE
+
+
+class DetConSTransform(MultiViewTransformV2):
+    """Implements the transformations for DetConS, based on SimCLR. [detcon]_ [simclr]_
```
[the remark this answers is not in the document]
Comment (liopeer) [reply lightly/transforms/detcon_transform.py]: fixed in https://github.com/lightly-ai/lightly/pull/1737/commits/f51e834d092463e4a3ca2595b3ca23537ff3ce5c

File: lightly/transforms/detcon_transform.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
+# from lightly.transforms import BYOLTransform
```
Comment (guarin) [inline lightly/transforms/detcon_transform.py:1]:
```suggestion
```

**Review by guarin — APPROVED** [review]
LGTM!
